### PR TITLE
Failed mass remove

### PR DESCRIPTION
**Pull request by 0xdiba**

### CODE REVIEW

File: lib/resque/failure/extra.rb [hunk 1]
```diff
@@ -114,6 +114,22 @@ def requeue_and_remove(index, queue=nil)
         mark_for_remove(index)
       end
 
+      # remove_older_than deletes all jobs older than the days provided
+      def remove_older_than(days)
```
Comment (astathopoulos) [inline lib/resque/failure/extra.rb]: I would add something like:
```
days = [30, days.to_i].min
```
to prevent removing all jobs by mistake.

Comment (0xdiba) [reply lib/resque/failure/extra.rb]: Do we want to restrict usage to such a long interval?
The refreshed UI takes care of some accidents by defaulting to 30 days.

File: lib/resque/failure/extra.rb [hunk 2]
```diff
@@ -24,15 +24,36 @@ def matching_job?(job, opts)
...
-            job[k.to_s]
-          end == v
+            job[k.to_s] == v
+          end
         }
       end
 
+      def job_older_than?(job, max_age)
+        now = DateTime.now
+        diff = now - DateTime.parse(job["failed_at"])
```
Comment (ctrochalakis) [inline lib/resque/failure/extra.rb]: What's the unit of that subtraction? Use that unit in the variable name, eg `diff_days`, so it's easier to follow.

File: lib/resque/failure/extra.rb [hunk 3]
```diff
@@ -24,15 +24,36 @@ def matching_job?(job, opts)
...
+            job_older_than?(job, v)
           else
-            job[k.to_s]
-          end == v
+            job[k.to_s] == v
+          end
         }
       end
 
+      def job_older_than?(job, max_age)
```
Comment (ctrochalakis) [inline lib/resque/failure/extra.rb]: `age` is a better name here.

File: lib/resque/failure/extra.rb [hunk 4]
```diff
@@ -24,15 +24,36 @@ def matching_job?(job, opts)
...
+        now = DateTime.now
+        diff = now - DateTime.parse(job["failed_at"])
+
+        unit = max_age[-1,1]
+        num = max_age.chop.to_i
+        $stdout.puts [(diff*24*60).to_i,unit,num].inspect
```
Comment (ctrochalakis) [inline lib/resque/failure/extra.rb]: We can drop that.

File: lib/resque/failure/extra.rb [hunk 5]
```diff
@@ -67,6 +88,8 @@ def each(*args,&block)
       def action_by(opts={}, &block)
         action = opts.delete(:action).to_sym || :requeue
         arg = opts.delete(:arg)
+        opts.delete(:older_than) if opts[:older_than].empty?
```
Comment (ctrochalakis) [inline lib/resque/failure/extra.rb]: We shouldn't treat `older_than` in a special way, drop all empty/nil value keys.

**Review by ctrochalakis — CHANGES_REQUESTED** [review]

File: lib/resque/failure/extra.rb [hunk 6]
```diff
@@ -24,15 +24,36 @@ def matching_job?(job, opts)
...
+            job[k.to_s] == v
+          end
         }
       end
 
+      def job_older_than?(job, age)
+        now = DateTime.now
+        diff_days = now - DateTime.parse(job["failed_at"])
+
+        unit = age[-1,1]
```
Comment (ctrochalakis) [inline lib/resque/failure/extra.rb]: `age[-1]`

**Review by ctrochalakis — APPROVED** [review]

**Pull request merged**